pvlib/pvsystem.py

```diff
@@ -898,6 +1091,282 @@ def __repr__(self):
             f'{attr}: {getattr(self, attr)}' for attr in attrs))
 
 
+class Array:
```

Comment:
FixedTiltArray?

Reply:
Or... split into `Array`, `FixedTiltArray(Array)`, `SingleAxisTrackingArray(Array)`? This makes more sense to me than maintaining `PVSystem`, `Array`, and `SingleAxisTracker(PVSystem)`. Basic idea:

```python
class FixedTiltArray(Array)
    """
    Parameters
    ----------
    surface_tilt: float or array-like, default 0
        Surface tilt angles in decimal degrees.
        The tilt angle is defined as degrees from horizontal
        (e.g. surface facing up = 0, surface facing horizon = 90)

    surface_azimuth: float or array-like, default 180
        Azimuth angle of the module surface.
        North=0, East=90, South=180, West=270.

    **kwargs
        Passed to Array. Or copy remainder of Array doc string to be explicit.


# could be in pvsystem.py (module is gradually becoming just the objects) or could be in tracking.py
class SingleAxisTrackerArray(Array)
    """
    Parameters
    ----------
    axis_tilt : float, default 0
        The tilt of the axis of rotation (i.e, the y-axis defined by
        axis_azimuth) with respect to horizontal, in decimal degrees.

    etc.

    **kwargs
        Passed to Array. Or copy remainder of Array doc string to be explicit.
```

I believe the only major challenge is that the `get_aoi` and `get_irradiance` methods would either need to differ in signature (as they do now, and thus present a challenge to a `PVSystem` wrapper) or in implementation (tracker methods would include a call to `singleaxis`, and thus would be less efficient in some workflows).

And finally deprecate old `SingleAxisTracking` class.

Reply:
I really like this, but I think it might be too much for this PR. I think differentiating between tracking/fixed at the `Array` level will provide much needed flexibility.

I would be inclined towards a uniform signature even if it means some workflows would suffer a performance hit. With a uniform signature It would be possible for users to define custom `Array` types that could work differently and avoid the inefficiency at the cost of generality.

Reply:
Ok to skip in this PR, but let's pick it up again in #1109 before we release 0.9.